library/core/src/ptr/const_ptr.rs

```diff
@@ -906,6 +933,50 @@ impl<T> *const [T] {
         // SAFETY: the caller ensures that `self` is dereferencable and `index` in-bounds.
         unsafe { index.get_unchecked(self) }
     }
+
+    /// Returns `None` if the pointer is null, or else returns a shared reference to
+    /// the value wrapped in `Some`.
+    ///
+    /// # Safety
+    ///
+    /// Behavior is undefined if any of the following conditions are violated:
+    ///
+    /// * The pointer must be [valid] for reads for `ptr.len() * mem::size_of::<T>()` many bytes,
+    ///   and it must be properly aligned. This means in particular:
```

Comment:
The requirement for `as_uninit_ref` and `as_uninit_slice` is basically the same -- aligned and dereferencable -- but it is stated very differently. I am not sure what is better, but this is a curious inconsistency.

Reply:
I personally like the wording of `as_uninit_slice` better. It is easier to skim a list of bullet points than to read a coherent text.

Reply:
Agreed. That's also how [`ptr:.write`](https://doc.rust-lang.org/nightly/std/ptr/fn.write.html) and similar methods look.

Reply:
Do you want me to change the docs for `as_ref` and `as_mut`, too? Then they are unified.

Reply:
That would be great if you could do that. :)

Reply:
I think I have everything added. Please review the safety clauses carefully, let's don't risk I forgot something :)